### PR TITLE
feat!: Refactor online stores by removing the contrib directory. (Breaking imports)

### PR DESCRIPTION
**Pull request by HaoXuAI**
# What this PR does / why we need it:

# Which issue(s) this PR fixes:

# Misc

### CODE REVIEW

**Review by franciscojavierarceo — COMMENTED** [review]
Looks like integration tests are failing?

**Comment by HaoXuAI** [comment]
> Looks like integration tests are failing?

seems passed now

**Comment by franciscojavierarceo** [comment]
This will break any imports users are doing with custom code, can you update or title?

**Comment by HaoXuAI** [comment]
> This will break any imports users are doing with custom code, can you update or title?

There should be non-breaking changes, I also added the old custom type as well.

**Comment by dmartinol** [comment]
> > This will break any imports users are doing with custom code, can you update or title?
> 
> There should be non-breaking changes, I also added the old custom type as well.

I think he's saying that any feature store definition importing the legacy modules would be broken, e.g.:
```py
>> import feast.infra.online_stores.contrib.hazelcast_online_store.hazelcast_online_store
...
ModuleNotFoundError: No module named 'feast.infra.online_stores.contrib.hazelcast_online_store.hazelcast_online_store'
```
Why don't you create aliases instead, with a warning to inform the user that these will be removed later? 
```py
# feast/infra/online_stores/contrib/qdrant.py

import warnings
from feast.infra.online_stores.cqdrant import QdrantOnlineStore

warnings.warn(
    "The module feast.infra.online_stores.contrib.qdrant is deprecated and will be removed in a future release. "
    "Please use feast.infra.online_stores.cqdrant instead.",
    DeprecationWarning,
    stacklevel=2,
)
```

File: sdk/python/feast/repo_config.py [hunk 1]
```diff
@@ -51,23 +51,36 @@
...
+    "mysql": "feast.infra.online_stores.mysql_online_store.mysql.MySQLOnlineStore",
+    "hazelcast": "feast.infra.online_stores.hazelcast_online_store.hazelcast_online_store.HazelcastOnlineStore",
+    "ikv": "feast.infra.online_stores.ikv_online_store.ikv.IKVOnlineStore",
+    "elasticsearch": "feast.infra.online_stores.elasticsearch_online_store.ElasticSearchOnlineStore",
     "remote": "feast.infra.online_stores.remote.RemoteOnlineStore",
-    "singlestore": "feast.infra.online_stores.contrib.singlestore_online_store.singlestore.SingleStoreOnlineStore",
-    "qdrant": "feast.infra.online_stores.contrib.qdrant.QdrantOnlineStore",
+    "singlestore": "feast.infra.online_stores.singlestore_online_store.singlestore.SingleStoreOnlineStore",
+    "qdrant": "feast.infra.online_stores.cqdrant.QdrantOnlineStore",
+    **LEGACY_ONLINE_STORE_CLASS_FOR_TYPE,
```
Comment (dmartinol) [inline sdk/python/feast/repo_config.py:83]: IIUC, `LEGACY_ONLINE_STORE_CLASS_FOR_TYPE` would never be used unless someone declares a store like:
```yaml
online_store:
  type: feast.infra.online_stores.contrib.qdrant.QdrantOnlineStore
  ...
```
Is this what you had in mind?

Comment (tokoko) [reply sdk/python/feast/repo_config.py:83]: agreed, I doubt anyone actually did anything like that. This is technically breaking, but I'm fine with it even w/o leaving aliases.

Comment (HaoXuAI) [reply sdk/python/feast/repo_config.py:83]: Right that's what I'm referring to. I remember there was somewhere in doc said you can config online_store with something like: `ype: feast.infra.online_stores.contrib.qdrant.QdrantOnlineStore`, but not sure if that's still the case.

Comment (HaoXuAI) [reply sdk/python/feast/repo_config.py:83]: Anyway let me add the warning to ease the issue.

**Comment by HaoXuAI** [comment]
> > > This will break any imports users are doing with custom code, can you update or title?
> > 
> > 
> > There should be non-breaking changes, I also added the old custom type as well.
> 
> I think he's saying that any feature store definition importing the legacy modules would be broken, e.g.:
> 
> ```python
> >> import feast.infra.online_stores.contrib.hazelcast_online_store.hazelcast_online_store
> ...
> ModuleNotFoundError: No module named 'feast.infra.online_stores.contrib.hazelcast_online_store.hazelcast_online_store'
> ```
> 
> Why don't you create aliases instead, with a warning to inform the user that these will be removed later?
> 
> ```python
> # feast/infra/online_stores/contrib/qdrant.py
> 
> import warnings
> from feast.infra.online_stores.cqdrant import QdrantOnlineStore
> 
> warnings.warn(
>     "The module feast.infra.online_stores.contrib.qdrant is deprecated and will be removed in a future release. "
>     "Please use feast.infra.online_stores.cqdrant instead.",
>     DeprecationWarning,
>     stacklevel=2,
> )
> ```

Creating a alias means I have to keep two folders for same store. I think that's more confusing.
Since the online_store are not meant to be used by user to directly import the module, I think it's ok to simply remove the old ones.

**Comment by HaoXuAI** [comment]
@franciscojavierarceo @dmartinol fixed the conflict after new online store, need another review.

**Review by franciscojavierarceo — APPROVED** [review]

**Pull request merged**